src/fsharp/vs/IncrementalBuild.fs

```diff
@@ -1163,6 +1166,26 @@ type IncrementalBuilder(frameworkTcImportsCache: FrameworkImportsCache, tcConfig
         let flags = tcConfig.ComputeCanContainEntryPoint(sourceFiles |> List.map snd)
         (sourceFiles,flags) ||> List.map2 (fun (m,nm) flag -> (m,nm,flag))
 
+    let defaultTimeStamp = DateTime.Now
+    let getFileTimeStamp (cache: TimeStampCache) fileName = 
+        let ok, v = cache.Files.TryGetValue(fileName)
```

Comment:
Just curious: why do you not use `match cache.Files.TryGetValue(fileName) with true, v -> ...`? Is your code more performant?